Web Inspector: move PageConsoleClient to FrameConsoleClient in preparation for Site Isolation

burg opened the pull request:
#### 462503322444ad95b557bdee4b9d99c6718976a3
<pre>
Web Inspector: move PageConsoleClient to FrameConsoleClient in preparation for Site Isolation
<a href="https://bugs.webkit.org/show_bug.cgi?id=298543">https://bugs.webkit.org/show_bug.cgi?id=298543</a>
<a href="https://rdar.apple.com/160131751">rdar://160131751</a>

Reviewed by Devin Rousso.

The current implementation of console logging in Web Inspector is broken under site isolation because
the inspector frontend is not made aware of cross-site iframes that were created in a
new web content process. Without a frontend connected, `InstrumentingAgents` in the new process
will not handle most instrumentation calls to the various Agents unless the agent is enabled.

Console domain events (such as `Console.messageAdded`) are not received because the `Console.enable`
command is never processed by the ConsoleAgent in this new web content process, which normally is sent
right after a frontend connects. Even without enablement, ConsoleAgent will buffer up to
`maximumConsoleMessages` (currently 100) without a frontend connection and then drop them on the floor.

Web Inspector needs to be able to connect to multiple web content processes in order to inspect
cross-origin iframes. In the case of ConsoleAgent, we want every frame to be able to log to console
regardless of which web process it is ultimately hosted in. That means there needs to be a ConsoleAgent
instance for each frame. Upon connecting to the new web content process, the frontend will drain
any buffered console messages.

To make this per-frame tracking possible, we are separately adding a new &apos;Frame&apos; target type to the Target
domain, to notify the frontend about Frame target lifecycle events. Using a separate Frame target means
that we can classify existing web instrumenting agents as &apos;Frame&apos; or &apos;Page&apos; based on how many instances we
expect to exist when Site Isolation is enabled. Agents with multiple instances should stay in WebProcess
and move from being Page-based to being LocalFrame-based. Agents with a single instance are those that
provide data collated from multiple iframes, such as Page, Network, and Storage. These agents will need to
move to UIProcess.

As a first step towards frame-based agents, move away from page-based console logging and instead
log messages through the relevant document&apos;s frame. These changes are needed to accomplish this:

- store a WeakRef to a Frame rather than a Page.
- store FrameConsoleClient on both LocalFrame and RemoteFrame.
- bail out if the Frame WeakRef went away or if the associated frame is remote.
- move consoleMessageListener (a test-only fixture) to Page so it is accessible from local subframes.

No new tests, no changes to functionality expected. This paves the way for refactoring ConsoleAgent to
be frame-based.

* LayoutTests/security/contentSecurityPolicy/plugins-types-allows-youtube-plugin-replacement.html:
Revert to not depending on a specific console message in order for the test to finish. This test
creates a subframe as part of plugin replacement, which caused us to create more than one FrameConsoleClient.
In the future we should consider removing consoleMessageListener entirely and using something more flexible.

* Source/WebCore/Headers.cmake:
* Source/WebCore/Modules/applepay/ApplePaySession.cpp:
* Source/WebCore/SaferCPPExpectations/UncheckedCallArgsCheckerExpectations:
* Source/WebCore/SaferCPPExpectations/UncheckedLocalVarsCheckerExpectations:
* Source/WebCore/SaferCPPExpectations/UncountedCallArgsCheckerExpectations:
* Source/WebCore/SaferCPPExpectations/UncountedLocalVarsCheckerExpectations:
* Source/WebCore/Sources.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/bindings/js/ScriptCachedFrameData.cpp:
(WebCore::ScriptCachedFrameData::restore):
* Source/WebCore/bindings/js/ScriptController.cpp:
(WebCore::ScriptController::initScriptForWindowProxy):
* Source/WebCore/bindings/js/WindowProxy.cpp:
(WebCore::WindowProxy::setDOMWindow):
* Source/WebCore/css/StyleSheetContents.cpp:
(WebCore::StyleSheetContents::parseAuthorStyleSheet):
* Source/WebCore/dom/Document.cpp:
(WebCore::Document::addConsoleMessage):
(WebCore::Document::addMessage):
* Source/WebCore/dom/ScriptableDocumentParser.h:
* Source/WebCore/inspector/InspectorInstrumentation.h:
(WebCore::InspectorInstrumentation::addMessageToConsole):
(WebCore::InspectorInstrumentation::consoleCount):
(WebCore::InspectorInstrumentation::consoleCountReset):
(WebCore::InspectorInstrumentation::takeHeapSnapshot):
(WebCore::InspectorInstrumentation::startConsoleTiming):
(WebCore::InspectorInstrumentation::logConsoleTiming):
(WebCore::InspectorInstrumentation::stopConsoleTiming):
(WebCore::InspectorInstrumentation::consoleTimeStamp):
(WebCore::InspectorInstrumentation::startProfiling):
(WebCore::InspectorInstrumentation::stopProfiling):
* Source/WebCore/inspector/agents/page/PageAuditAgent.cpp:
(WebCore::PageAuditAgent::muteConsole):
(WebCore::PageAuditAgent::unmuteConsole):
* Source/WebCore/inspector/agents/page/PageDebuggerAgent.cpp:
(WebCore::PageDebuggerAgent::muteConsole):
(WebCore::PageDebuggerAgent::unmuteConsole):
(WebCore::PageDebuggerAgent::breakpointActionLog):
* Source/WebCore/inspector/agents/page/PageNetworkAgent.cpp:
(WebCore::PageNetworkAgent::addConsoleMessage):
* Source/WebCore/inspector/agents/page/PageRuntimeAgent.cpp:
(WebCore::PageRuntimeAgent::muteConsole):
(WebCore::PageRuntimeAgent::unmuteConsole):
* Source/WebCore/loader/ResourceLoader.cpp:
(WebCore::ResourceLoader::didReceiveResponse):
* Source/WebCore/page/DOMWindow.cpp:
(WebCore::DOMWindow::console const):
(WebCore::DOMWindow::checkedConsole const):
(WebCore::DOMWindow::printErrorMessage const):
(WebCore::DOMWindow::isInsecureScriptAccess):
* Source/WebCore/page/DOMWindow.h:
* Source/WebCore/page/Frame.cpp:
(WebCore::Frame::Frame):
* Source/WebCore/page/Frame.h:
(WebCore::Frame::console):
(WebCore::Frame::console const):
* Source/WebCore/page/FrameConsoleClient.cpp: Renamed from Source/WebCore/page/PageConsoleClient.cpp.
(WebCore::FrameConsoleClient::FrameConsoleClient):
(WebCore::FrameConsoleClient::shouldPrintExceptions):
(WebCore::FrameConsoleClient::setShouldPrintExceptions):
(WebCore::FrameConsoleClient::mute):
(WebCore::FrameConsoleClient::unmute):
(WebCore::FrameConsoleClient::logMessageToSystemConsole):
(WebCore::FrameConsoleClient::addMessage):
(WebCore::FrameConsoleClient::messageWithTypeAndLevel):
(WebCore::FrameConsoleClient::count):
(WebCore::FrameConsoleClient::countReset):
(WebCore::FrameConsoleClient::profile):
(WebCore::FrameConsoleClient::profileEnd):
(WebCore::FrameConsoleClient::takeHeapSnapshot):
(WebCore::FrameConsoleClient::time):
(WebCore::FrameConsoleClient::timeLog):
(WebCore::FrameConsoleClient::timeEnd):
(WebCore::FrameConsoleClient::timeStamp):
(WebCore::FrameConsoleClient::record):
(WebCore::FrameConsoleClient::recordEnd):
(WebCore::FrameConsoleClient::screenshot):
* Source/WebCore/page/FrameConsoleClient.h: Renamed from Source/WebCore/page/PageConsoleClient.h.
* Source/WebCore/page/IntersectionObserver.cpp:
Add missing header caused by changes to UnifiedSources segmentation.
* Source/WebCore/page/LocalDOMWindow.cpp:
(WebCore::LocalDOMWindow::processPostMessage):
* Source/WebCore/page/Page.cpp:
(WebCore::Page::Page):
(WebCore::Page::setConsoleMessageListenerForTesting):
(WebCore::Page::consoleMessageListenerForTesting const):
(WebCore::Page::injectUserStyleSheet):
* Source/WebCore/page/Page.h:
(WebCore::Page::console): Deleted.
(WebCore::Page::console const): Deleted.
* Source/WebCore/testing/Internals.cpp:
(WebCore::Internals::resetToConsistentState):
(WebCore::Internals::setConsoleMessageListener):
* Source/WebCore/testing/Internals.h:
* Source/WebCore/workers/WorkerGlobalScope.cpp:
(WebCore::WorkerGlobalScope::addConsoleMessage):
* Source/WebCore/worklets/WorkletGlobalScope.cpp:
(WebCore::WorkletGlobalScope::logExceptionToConsole):
(WebCore::WorkletGlobalScope::addConsoleMessage):
* Source/WebCore/xml/XSLStyleSheetLibxslt.cpp:
(WebCore::XSLStyleSheet::parseString):
* Source/WebCore/xml/XSLTProcessorLibxslt.cpp:
(WebCore::XSLTProcessor::parseErrorFunc):
(WebCore::docLoaderFunc):
* Source/WebCore/xml/parser/XMLDocumentParserLibxml2.cpp:
(WebCore::openFunc):
* Source/WebKitLegacy/mac/Misc/WebCoreStatistics.mm:
(+[WebCoreStatistics shouldPrintExceptions]):
(+[WebCoreStatistics setShouldPrintExceptions:]):

Canonical link: <a href="https://commits.webkit.org/299982@main">https://commits.webkit.org/299982@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/22c808bf0390f75a1598edf5eb5af6a61a7c3292

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/120970 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/40666 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/31324 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/127385 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/73050 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/ad6d3e10-31bf-49f3-9704-8ba5a7676267) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/122846 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/41368 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/49245 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/91885 "Passed tests") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/61129 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/86815f79-6455-4940-8539-fbf65cb28965) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/123922 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/33032 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/108445 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/72573 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/b08312a9-e262-4147-aa87-3e7fd9ab609e) 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/133/builds/32060 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/26546 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/70974 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/102535 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/26727 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/130242 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/47897 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/36395 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/100502 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/48265 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/104612 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/100404 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/25446 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/45802 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/23847 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/44585 "Built successfully") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/47755 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/53468 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/47226 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/50573 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/48910 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

dcrousso commented on Source/WebCore/page/FrameConsoleClient.h:
```diff
@@ -94,10 +94,8 @@ class WEBCORE_EXPORT PageConsoleClient final : public JSC::ConsoleClient, public
-
-    WeakRef<Page> m_page;
     RefPtr<StringCallback> m_consoleMessageListener;
+    WeakRef<Frame> m_frame;
```
it seems like pretty much every use of this is actually expecting a `LocalFrame`.  are we using just `Frame` here on the expectation that we'll eventually want to support `RemoteFrame` too?  i thought that each WebProcess would have their own set of agents so it would only need to be concerned with stuff happening in that WebProcess (i.e. each WebProcess would directly send it's messages to Web Inspector instead of proxying from one WebProcess to another until the top frame is reached)

burg replied:
Yes, it doesn't do anything if Frame is actually a RemoteFrame, only if its a LocalFrame. The use of `Frame` type instead of `LocalFrame` could be transitional until more callers use LocalFrame in the code that does the actual logging. Let's revisit this in a while.

dcrousso commented on Source/WebCore/bindings/js/WindowProxy.cpp:
```diff
@@ -199,11 +199,10 @@ void WindowProxy::setDOMWindow(DOMWindow* newDOMWindow)
             cacheableBindingRootObject->updateGlobalObject(windowProxy->window());
 
+        windowProxy->window()->setConsoleClient(m_frame->console());
         windowProxy->attachDebugger(page ? page->debugger() : nullptr);
```
Aside: we'll eventually need to change this too, right?  i.e. `PageDebugger` -> `FrameDebugger`

dcrousso reviewed (approved):
neat!

burg committed:
Web Inspector: move PageConsoleClient to FrameConsoleClient in preparation for Site Isolation
https://bugs.webkit.org/show_bug.cgi?id=298543
rdar://160131751

Reviewed by Devin Rousso.

The current implementation of console logging in Web Inspector is broken under site isolation because
the inspector frontend is not made aware of cross-site iframes that were created in a
new web content process. Without a frontend connected, `InstrumentingAgents` in the new process
will not handle most instrumentation calls to the various Agents unless the agent is enabled.

Console domain events (such as `Console.messageAdded`) are not received because the `Console.enable`
command is never processed by the ConsoleAgent in this new web content process, which normally is sent
right after a frontend connects. Even without enablement, ConsoleAgent will buffer up to
`maximumConsoleMessages` (currently 100) without a frontend connection and then drop them on the floor.

Web Inspector needs to be able to connect to multiple web content processes in order to inspect
cross-origin iframes. In the case of ConsoleAgent, we want every frame to be able to log to console
regardless of which web process it is ultimately hosted in. That means there needs to be a ConsoleAgent
instance for each frame. Upon connecting to the new web content process, the frontend will drain
any buffered console messages.

To make this per-frame tracking possible, we are separately adding a new 'Frame' target type to the Target
domain, to notify the frontend about Frame target lifecycle events. Using a separate Frame target means
that we can classify existing web instrumenting agents as 'Frame' or 'Page' based on how many instances we
expect to exist when Site Isolation is enabled. Agents with multiple instances should stay in WebProcess
and move from being Page-based to being LocalFrame-based. Agents with a single instance are those that
provide data collated from multiple iframes, such as Page, Network, and Storage. These agents will need to
move to UIProcess.

As a first step towards frame-based agents, move away from page-based console logging and instead
log messages through the relevant document's frame. These changes are needed to accomplish this:

- store a WeakRef to a Frame rather than a Page.
- store FrameConsoleClient on both LocalFrame and RemoteFrame.
- bail out if the Frame WeakRef went away or if the associated frame is remote.
- move consoleMessageListener (a test-only fixture) to Page so it is accessible from local subframes.

No new tests, no changes to functionality expected. This paves the way for refactoring ConsoleAgent to
be frame-based.

* LayoutTests/security/contentSecurityPolicy/plugins-types-allows-youtube-plugin-replacement.html:
Revert to not depending on a specific console message in order for the test to finish. This test
creates a subframe as part of plugin replacement, which caused us to create more than one FrameConsoleClient.
In the future we should consider removing consoleMessageListener entirely and using something more flexible.

* Source/WebCore/Headers.cmake:
* Source/WebCore/Modules/applepay/ApplePaySession.cpp:
* Source/WebCore/SaferCPPExpectations/UncheckedCallArgsCheckerExpectations:
* Source/WebCore/SaferCPPExpectations/UncheckedLocalVarsCheckerExpectations:
* Source/WebCore/SaferCPPExpectations/UncountedCallArgsCheckerExpectations:
* Source/WebCore/SaferCPPExpectations/UncountedLocalVarsCheckerExpectations:
* Source/WebCore/Sources.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/bindings/js/ScriptCachedFrameData.cpp:
(WebCore::ScriptCachedFrameData::restore):
* Source/WebCore/bindings/js/ScriptController.cpp:
(WebCore::ScriptController::initScriptForWindowProxy):
* Source/WebCore/bindings/js/WindowProxy.cpp:
(WebCore::WindowProxy::setDOMWindow):
* Source/WebCore/css/StyleSheetContents.cpp:
(WebCore::StyleSheetContents::parseAuthorStyleSheet):
* Source/WebCore/dom/Document.cpp:
(WebCore::Document::addConsoleMessage):
(WebCore::Document::addMessage):
* Source/WebCore/dom/ScriptableDocumentParser.h:
* Source/WebCore/inspector/InspectorInstrumentation.h:
(WebCore::InspectorInstrumentation::addMessageToConsole):
(WebCore::InspectorInstrumentation::consoleCount):
(WebCore::InspectorInstrumentation::consoleCountReset):
(WebCore::InspectorInstrumentation::takeHeapSnapshot):
(WebCore::InspectorInstrumentation::startConsoleTiming):
(WebCore::InspectorInstrumentation::logConsoleTiming):
(WebCore::InspectorInstrumentation::stopConsoleTiming):
(WebCore::InspectorInstrumentation::consoleTimeStamp):
(WebCore::InspectorInstrumentation::startProfiling):
(WebCore::InspectorInstrumentation::stopProfiling):
* Source/WebCore/inspector/agents/page/PageAuditAgent.cpp:
(WebCore::PageAuditAgent::muteConsole):
(WebCore::PageAuditAgent::unmuteConsole):
* Source/WebCore/inspector/agents/page/PageDebuggerAgent.cpp:
(WebCore::PageDebuggerAgent::muteConsole):
(WebCore::PageDebuggerAgent::unmuteConsole):
(WebCore::PageDebuggerAgent::breakpointActionLog):
* Source/WebCore/inspector/agents/page/PageNetworkAgent.cpp:
(WebCore::PageNetworkAgent::addConsoleMessage):
* Source/WebCore/inspector/agents/page/PageRuntimeAgent.cpp:
(WebCore::PageRuntimeAgent::muteConsole):
(WebCore::PageRuntimeAgent::unmuteConsole):
* Source/WebCore/loader/ResourceLoader.cpp:
(WebCore::ResourceLoader::didReceiveResponse):
* Source/WebCore/page/DOMWindow.cpp:
(WebCore::DOMWindow::console const):
(WebCore::DOMWindow::checkedConsole const):
(WebCore::DOMWindow::printErrorMessage const):
(WebCore::DOMWindow::isInsecureScriptAccess):
* Source/WebCore/page/DOMWindow.h:
* Source/WebCore/page/Frame.cpp:
(WebCore::Frame::Frame):
* Source/WebCore/page/Frame.h:
(WebCore::Frame::console):
(WebCore::Frame::console const):
* Source/WebCore/page/FrameConsoleClient.cpp: Renamed from Source/WebCore/page/PageConsoleClient.cpp.
(WebCore::FrameConsoleClient::FrameConsoleClient):
(WebCore::FrameConsoleClient::shouldPrintExceptions):
(WebCore::FrameConsoleClient::setShouldPrintExceptions):
(WebCore::FrameConsoleClient::mute):
(WebCore::FrameConsoleClient::unmute):
(WebCore::FrameConsoleClient::logMessageToSystemConsole):
(WebCore::FrameConsoleClient::addMessage):
(WebCore::FrameConsoleClient::messageWithTypeAndLevel):
(WebCore::FrameConsoleClient::count):
(WebCore::FrameConsoleClient::countReset):
(WebCore::FrameConsoleClient::profile):
(WebCore::FrameConsoleClient::profileEnd):
(WebCore::FrameConsoleClient::takeHeapSnapshot):
(WebCore::FrameConsoleClient::time):
(WebCore::FrameConsoleClient::timeLog):
(WebCore::FrameConsoleClient::timeEnd):
(WebCore::FrameConsoleClient::timeStamp):
(WebCore::FrameConsoleClient::record):
(WebCore::FrameConsoleClient::recordEnd):
(WebCore::FrameConsoleClient::screenshot):
* Source/WebCore/page/FrameConsoleClient.h: Renamed from Source/WebCore/page/PageConsoleClient.h.
* Source/WebCore/page/IntersectionObserver.cpp:
Add missing header caused by changes to UnifiedSources segmentation.
* Source/WebCore/page/LocalDOMWindow.cpp:
(WebCore::LocalDOMWindow::processPostMessage):
* Source/WebCore/page/Page.cpp:
(WebCore::Page::Page):
(WebCore::Page::setConsoleMessageListenerForTesting):
(WebCore::Page::consoleMessageListenerForTesting const):
(WebCore::Page::injectUserStyleSheet):
* Source/WebCore/page/Page.h:
(WebCore::Page::console): Deleted.
(WebCore::Page::console const): Deleted.
* Source/WebCore/testing/Internals.cpp:
(WebCore::Internals::resetToConsistentState):
(WebCore::Internals::setConsoleMessageListener):
* Source/WebCore/testing/Internals.h:
* Source/WebCore/workers/WorkerGlobalScope.cpp:
(WebCore::WorkerGlobalScope::addConsoleMessage):
* Source/WebCore/worklets/WorkletGlobalScope.cpp:
(WebCore::WorkletGlobalScope::logExceptionToConsole):
(WebCore::WorkletGlobalScope::addConsoleMessage):
* Source/WebCore/xml/XSLStyleSheetLibxslt.cpp:
(WebCore::XSLStyleSheet::parseString):
* Source/WebCore/xml/XSLTProcessorLibxslt.cpp:
(WebCore::XSLTProcessor::parseErrorFunc):
(WebCore::docLoaderFunc):
* Source/WebCore/xml/parser/XMLDocumentParserLibxml2.cpp:
(WebCore::openFunc):
* Source/WebKitLegacy/mac/Misc/WebCoreStatistics.mm:
(+[WebCoreStatistics shouldPrintExceptions]):
(+[WebCoreStatistics setShouldPrintExceptions:]):

Canonical link: https://commits.webkit.org/299982@main
